pkg/controllers/request_controller.go

```
@@ -103,10 +103,15 @@ func (r *AccessRequestController) Create(c *gin.Context) {
 	}
 
 	// Save it to DB
-	Db.InsertAccessRequest(ctx, data)
+	if err := Db.InsertAccessRequest(ctx, data); err != nil {
+		c.AbortWithStatusJSON(errors.ErrorDatabaseInsert(err))
+		return
+	}
 
 	// Fire creation event
-	Event.AccessRequestCreated(ctx, data)
+	if err := Event.AccessRequestCreated(ctx, data); err != nil {
+		log.Error().Err(err).Msg("failed to fire AccessRequestCreated event")
+	}
```

Not sure if we should return an error here or of just logging is enough

---

I think logging is enough